stepconf/stepconf.go

```diff
@@ -102,9 +102,52 @@ func parseTag(tag string) (string, string) {
 	return tag, ""
 }
 
-// Parse populates a struct with the retrieved values from environment variables
-// described by struct tags and applies the defined validations.
-func Parse(conf interface{}) error {
+// EnvProvider ...
+type EnvProvider interface {
+	Getenv(key string) string
+}
+
+// OSEnvProvider ...
+type OSEnvProvider struct{}
+
+// NewOSEnvProvider ...
+func NewOSEnvProvider() EnvProvider {
+	return OSEnvProvider{}
+}
+
+// Getenv ...
+func (p OSEnvProvider) Getenv(key string) string {
+	return os.Getenv(key)
+}
+
+// EnvParser ...
+type EnvParser struct {
+	envProvider EnvProvider
+}
+
+// EnvParserOpts ...
+type EnvParserOpts struct {
+	EnvProvider EnvProvider
+}
+
+// NewDefaultEnvParser ...
+func NewDefaultEnvParser() EnvParser {
+	return NewEnvParser(
+		EnvParserOpts{
+			EnvProvider: NewOSEnvProvider(),
+		},
+	)
+}
+
+// NewEnvParser ...
+func NewEnvParser(opts EnvParserOpts) EnvParser {
```

Comment:
I'd rather just have the parameters as parameters. EnvParserOpts seems to be unnecessary. It is not really a "non-breaking" change (however go handles when a new field is not filled) because semantically when a new parameter is introduced it should be (at least considered) to be filled so this solution just hides the "breaking" change.

Reply:
EnvParserOpts should also be built via New*** method and there the breaking change is obvious